api-admin/src/main/java/bio/terra/pearl/api/admin/service/ConfigExtService.java

```diff
@@ -39,4 +46,33 @@ private Map<String, String> buildConfigMap() {
         "adminApiHostname",
             StringUtils.defaultIfEmpty(applicationRoutingPaths.getAdminApiHostname(), ""));
   }
+
+  /**
+   * returns non-public configuration information -- note that this still should not return actual
+   * secrets
+   */
+  public Map<String, ?> getInternalConfigMap(AdminUser user) {
+    if (!user.isSuperuser()) {
+      throw new PermissionDeniedException("You do not have permission to view this config");
+    }
+    var configMap =
+        Map.of(
+            "pepperDsmConfig",
+            Map.of(
+                "useLiveDsm",
+                pepperDSMConfig.useLiveDsm(),
+                "secret",
+                pepperDSMConfig.getSecret().substring(0, 4)
+                    + "..."
+                    + pepperDSMConfig
+                        .getSecret()
+                        .substring(
+                            pepperDSMConfig.getSecret().length() - 3,
+                            pepperDSMConfig.getSecret().length()),
```

Comment:
this assumes that this particular secret will never be rotated for a shorter secret, which could lead to this either revealing the entire secret, or revealing enough of it where it becomes easily guessable. which is probably fine to assume for now, but might not always be the case.

not a blocker and I know you just want it in order to debug live DSM right now, but since it's security related I wanted to at least mention the concern, especially since it might be more applicable to other config maps that we add to this endpoint in the future.

Reply:
yeah, hopefully we never have <10 character secrets, but I like the idea of being super safe